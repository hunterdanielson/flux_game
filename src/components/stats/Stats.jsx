import React from 'react';
import PropTypes from 'prop-types';

const Stats = ({ health, attack, defense }) => (
  <dl>
    <dt>Health</dt>
    <dd>{health}</dd>
    <dt>Attack</dt>
    <dd>{attack}</dd>
    <dt>Defense</dt>
    <dd>{defense}</dd>
  </dl>
);

Stats.propTypes = {
  health: PropTypes.number.isRequired,
  attack: PropTypes.number.isRequired,
  defense: PropTypes.number.isRequired
};

export default Stats;
