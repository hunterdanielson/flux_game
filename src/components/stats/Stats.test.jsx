import React from 'react';
import { shallow } from 'enzyme';
import Stats from './Stats';

describe('Stats component', () => {
  it('renders Stats', () => {
    const wrapper = shallow(<Stats health={100} attack={10} defense={10} />);
    expect(wrapper).toMatchSnapshot();
  });
});
