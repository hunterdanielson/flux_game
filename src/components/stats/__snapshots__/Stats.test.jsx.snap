// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats component renders Stats 1`] = `
<dl>
  <dt>
    Health
  </dt>
  <dd>
    100
  </dd>
  <dt>
    Attack
  </dt>
  <dd>
    10
  </dd>
  <dt>
    Defense
  </dt>
  <dd>
    10
  </dd>
</dl>
`;
